lib/controller.php

```diff
@@ -179,17 +181,28 @@ public function import_blob($blob) {
 			$body = wpmarkdown_markdown_to_html( $body );
 		}
 
+		$post = new WordPress_GitHub_Sync_Post( $args['ID'] );
+
 		// Can we really just mash everything together here?
+		$post_type = $post->get_type_from_path($path);
+		$post_name = $post->get_name_from_path($path);
```

Comment:
@mAAdhaTTah I understand, but my point was when you create a file within GitHub, you can just put a title and the content, omitting the `layout` and the plugin would determine which `post_type` is it based on the directory it has been stored.

`_posts = post_type post`, `_pages = post_type page`, `_books = post_type book`, so that way you don't mess up the structure by mistake, like put a `post` inside the  `page` layout/post_type directory.

I missed a condition where it was only to use that `post_type` only if there wasn't any declared in the YAML frontmatter.